### DISCUSSION TITLE
Docstring/help not showing as expected

### DISCUSSION THREAD

**Discussion by MatthiasKohl** [wjakob/nanobind#79] [General]
In my nanobind extension module, if I do:

```
import extension_module
help(extension_module.foo)
```

I get the following docstring for all my functions:

```
Help on nb_func in module cugraph_ops_nb object:

foo = class nb_func(object)
 |  nanobind function object
 |
 |  Methods defined here:
 |
 |  __call__(self, /, *args, **kwargs)
 |      Call self as a function.
 |
 |  __getattribute__(self, name, /)
 |      Return getattr(self, name).
 |
 |  ----------------------------------------------------------------------
 |  Static methods defined here:
 |
 |  __new__(*args, **kwargs) from builtins.type
 |      Create and return a new object.  See help(type) for accurate signature.
```

This is with the docstring added in the bindings.
I'm also quite sure that this has worked before, but the environment where I built the extension may have been different.

Does anyone know why I could be getting the above docstring ?

**Comment by wjakob** [comment]
Doesn't the nanobind11 test suite test docstrings for functions?

To me it looks like you are visualizing the docstring of the function's type (internal `nb_type` type object created by nanobind) instead of the function *instance*.

**Reply by MatthiasKohl** [reply]
I'm guessing that I'm doing something wrong here, but I can see that my functions have the right `__doc__` attribute, but using the python interpreter `help` on them doesn't work as expected.
Maybe you'd have to do something like this : 
```
>>> foo = extension_module.function
>>> help(foo)
```
but that seems odd.

**Reply by MatthiasKohl** [reply]
I don't know what the python interpreter `help` is actually doing though, so I don't know why I'm not simply getting the string from `__doc__`

**Reply by wjakob** [reply]
It turns out that this is was a regression -- it used to work. Some refactoring in `nb_func` caused very minor changes that caused pydoc to no longer inspect the function's docstrings. Fixed in https://github.com/wjakob/nanobind/commit/384f4ada1f3f08486fb03427227878ddbbcaad43.
